### PR TITLE
Introduce SQLClaim that can handle any RDS database engine

### PR DESCRIPTION
**Pull request by ytsarev**
### Description of your changes

Instead of hardcoded PostgreSQL, make generic SQL XRD/Composition that can handle arbitrary RDS instance engine configuration

* PostgrSQLInstance -> SQLInstance
* expose `engine` and `engineVersion` in XRD
* Add 3306 SecurityGroupRule
* Provide mariadb example

Signed-off-by: Yury Tsarev <yury@upbound.io>

I have:

- [x] Read and followed Upbound's [contribution process](https://git.io/fj2m9).
- [ ] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR, as appropriate.

### How has this code been tested

```
k apply -f examples/cluster-claim.yaml,examples/postgres-claim.yaml,examples/mariadb-claim.yaml
```

PostgreSQL connectivity test from within instantiated EKS cluster
```
cat psql.yaml
apiVersion: v1
kind: Pod
metadata:
  name: see-db
  namespace: default
spec:
  containers:
  - name: see-db
    image: postgres:12
    command: ['psql']
    args: ['-c', 'SELECT current_database();']
    env:
    - name: PGDATABASE
      value: postgres
    - name: PGHOST
      value: platform-ref-aws-db-postgres-t5jnc-hgktq.cxal1lomznba.us-west-2.rds.amazonaws.com
    - name: PGUSER
      value: masteruser
    - name: PGPASSWORD
      value: <password>

k logs -f see-db
 current_database
------------------
 postgres
(1 row)
```

mariadb connectivity test from within instantiated EKS cluster
```
kubectl run -i --tty --rm debug --image=mysql --restart=Never -- sh
sh-4.4# mysql -h platform-ref-aws-db-mariadb-grpz9-92kzs.cxal1lomznba.us-west-2.rds.amazonaws.com -u masteruser -p
Enter password:
Welcome to the MySQL monitor.  Commands end with ; or \g.
Your MySQL connection id is 79
Server version: 5.5.5-10.6.10-MariaDB managed by https://aws.amazon.com/rds/

Copyright (c) 2000, 2022, Oracle and/or its affiliates.

Oracle is a registered trademark of Oracle Corporation and/or its
affiliates. Other names may be trademarks of their respective
owners.

Type 'help;' or '\h' for help. Type '\c' to clear the current input statement.

mysql> show databases
    -> ;
+--------------------+
| Database           |
+--------------------+
| information_schema |
| innodb             |
| mysql              |
| performance_schema |
| sys                |
+--------------------+
5 rows in set (0.00 sec)
```

### CODE REVIEW

**Review by stevendborrelli — APPROVED** [review]
Approved with a few minor comments. 

```console
kubectl get -f postgres-claim.yaml 
NAME                                                                  SYNCED   READY   CONNECTION-SECRET                   AGE
sqlinstance.aws.platformref.upbound.io/platform-ref-aws-db-postgres   True     True    platform-ref-aws-db-conn-postgres   31m

NAME                TYPE     DATA   AGE
secret/psqlsecret   Opaque   1      31m

kubectl get -f mariadb-claim.yaml
NAME                                                                 SYNCED   READY   CONNECTION-SECRET                  AGE
sqlinstance.aws.platformref.upbound.io/platform-ref-aws-db-mariadb   True     True    platform-ref-aws-db-conn-mariadb   4m10s
```


- directory `database/postgres` be something like `database/sql`
- The readme should indicate that the cluster is required to provision the DBs. In the graph in the readme they look like independent paths.

File: package/database/postgres/definition.yaml [hunk 1]
```diff
@@ -29,6 +29,12 @@ spec:
               parameters:
                 type: object
                 properties:
+                  engine:
```
[the remark this answers is not in the document]
Comment (ytsarev) [reply package/database/postgres/definition.yaml:32]: I was thinking about it but do not want to immediately maintain everything that is described in https://docs.aws.amazon.com/AmazonRDS/latest/APIReference/API_CreateDBInstance.html

**Comment by ytsarev** [comment]
@stevendborrelli PR amended, could you please take another brief look?

**Review by stevendborrelli — APPROVED** [review]
LGTM, cluster + mariadb provisioned successfully.

**Pull request merged**